app/eventPage/[id].tsx

```diff
@@ -72,17 +93,48 @@ export default function EventPage() {
           )}
         </YStack>
       </ScrollView>
-      {(typeof event.organizer !== "number" &&
-        user?.id === event.organizer.id) || (
-        <FloatingActionButton
-          iconAfter={
-            <IconSymbol name="square.and.pencil" size={24} color="$color8" />
-          }
-          // onPress={() => navigation.navigate("eventEdit", { eventId: event.docs[0].id })}
-        >
-          Edit Event
-        </FloatingActionButton>
-      )}
+      {joinStatus !== "joined" &&
+        typeof event.organizer !== "number" &&
+        user?.id === event.organizer.id && (
+          <FloatingActionButton
+            iconAfter={
+              <IconSymbol name="square.and.pencil" size={24} color="$color8" />
+            }
+          >
+            Edit Event
+          </FloatingActionButton>
+        )}
+      {joinStatus !== "joined" &&
+        typeof event.organizer !== "number" &&
+        user?.id !== event.organizer.id && (
+          <Link
```

Comment:
This button renders weirdly

<img width="555" alt="image" src="https://github.com/user-attachments/assets/77ed2b6c-4298-46ac-8d36-9b26c5159f7e">

I think I know why though - this link is missing an `asChild` prop which would make it so that the button inside is the link instead of the link containing the button (and yeah, I know, that's how it is supposed to work on the web but RN is somewhat different in that aspect I suppose)